kroxylicious-integration-test-support/src/main/java/io/kroxylicious/test/tester/SimpleMetric.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.test.tester;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Simple representation of a Prometheus metric
+ * @param name metric name
+ * @param labels metric labels
+ * @param value metric value
+ */
+public record SimpleMetric(String name, Map<String, String> labels, double value) {
+
+    // https://github.com/prometheus/docs/blob/main/content/docs/instrumenting/exposition_formats.md
+    // note: RE doesn't handle escaping within label values
+    @SuppressWarnings("java:S5852") //
+    private static final Pattern PROM_TEXT_EXPOSITION_PATTERN = Pattern
+            .compile("^(?<metric>[a-zA-Z_:][a-zA-Z0-9_:]*)(\\{(?<labels>.*)})?[\\t ]*(?<value>[0-9E.]*)[\\t ]*(?<timestamp>\\d+)?$");
+    private static final Pattern NAME_WITH_QUOTED_VALUE = Pattern.compile("^(?<name>[a-zA-Z_:][a-zA-Z0-9_:]*)=\"(?<value>.*)\"$");
+
+    static List<SimpleMetric> parse(String output) {
+        var all = new ArrayList<SimpleMetric>();
+        try (var reader = new BufferedReader(new StringReader(output))) {
+            var line = reader.readLine();
```

Comment:
BufferedReader has a lines method that returns Stream<String>. I've pushed up a PR https://github.com/k-wall/kroxylicious/pull/17 refactoring so we don't have to work with null checks and readLine